e2e/testcafe-devextreme/tests/dataGrid/keyboardNavigation/keyboardNavigation.functional.ts

```diff
@@ -4629,3 +4629,62 @@ test('TreeList/DataGrid - Focus indicator is not visible when the Toolbar includ
   keyExpr: 'field_0',
   showBorders: true,
 }));
+
+test('Enter key should not trigger other function besides the function assigned on the clicked button', async (t) => {
+  const dataGrid = new DataGrid('#container');
+
+  await t
+    .click(dataGrid.getDataCell(0, 0).element)
+    .click(Selector('#editButton'))
+    .pressKey('tab')
+    .pressKey('enter');
+
+  const message = await Selector('#otherContainer').innerText;
+  const focusButtonMessage = message.includes(
+    'second button',
+  );
+
+  await t
+    .expect(focusButtonMessage)
+    .ok();
+}).before(async () => createWidget('dxDataGrid', {
+  dataSource: [
+    {
+      ID: 1,
+      Prefix: 'Mr.',
+      FirstName: 'John',
+    },
+  ],
+  keyExpr: 'ID',
+  columns: [
+    {
+      dataField: 'Prefix',
+      caption: 'Title',
+      width: 70,
+    },
+    'FirstName',
+  ],
```

Comment:
Let's simplify the ```DataGrid``` configuration a little:

```suggestion
  columns: [ 'Prefix', 'FirstName' ],
```